src/app/core/client-module/cyber-skills.guard.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  Router,
+  RouterStateSnapshot,
+  UrlTree,
+} from '@angular/router';
+import { AuthService } from '../auth-module/auth.service';
+import { CookieService } from 'ngx-cookie-service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CyberSkillsGuard implements CanActivate {
+  constructor(
+    private router: Router,
+    private auth: AuthService,
+    private cookies: CookieService,
+  ) {}
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot,
+  ): Promise<boolean | UrlTree> | boolean {
+    const c = this.cookies.get('presence');
+
+    if (c) {
+      return this.auth.validateToken().then(
+        (val) => {
+          return (
+            this.auth.accessGroups.includes('admin') ||
+            this.auth.accessGroups.includes('editor') ||
+            this.auth.accessGroups.includes('curator')
+          );
```

Comment:
just making sure, it shouldn't check for 'curator@cyberskills2work'?

Reply:
yes, updated :)